src/serac/physics/solid_mechanics.hpp

```diff
@@ -744,7 +751,11 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
     // Add the sign correction to move the term to the RHS
     adjoint_load_vector *= -1.0;
 
-    auto& lin_solver = nonlin_solver_.LinearSolver();
+    auto* lin_solver = nonlin_solver_->LinearSolver();
+
+    SLIC_ERROR_ROOT_IF(
+        !lin_solver,
+        "Adjoint solves currently need a linear solver associated with the supplied forward nonlinear solver");
```

Comment:
Any thoughts @btalamini? I know you were interested in implementing custom nonlinear solvers that may or may not have internal linear solvers. My thinking is that you can still define a custom `EquationSolver` with appropriate linear (adjoint) solvers in that case. We just need to make sure it has been defined. I guess the big question is do we want to allow equation solvers that do not have linear solvers? If not, I can ensure that the given pointer in the `EquationSolver` constructor is non-null and just return a reference here.

Reply:
I don't want to over-design it. I think it's ok to require that a nonlinear solver has to have a linear solver. We can reorganize later if necessary. There is a very strong incentive to re-use the linear solver (since the martix is essentially the same) that it makes sense to push the user in this direction.